test/auto/injectorSpec.js

```diff
@@ -245,12 +245,25 @@ describe('injector', function() {
 
     // Only Chrome and Firefox support this syntax.
     if (/chrome|firefox/i.test(navigator.userAgent)) {
-      it('should be possible to annotate functions that are declared using ES6 syntax', function() {
-        /*jshint -W061 */
-        // The function is generated using `eval` as just having the ES6 syntax can break some browsers.
-        expect(annotate(eval('({ fn(x) { return; } })').fn)).toEqual(['x']);
-        /*jshint +W061 */
+      /*jshint -W061 */
+      describe('ES6 syntax', function() {
+        // Functions and classes are generated using `eval` as just having the ES6 syntax can break
+        // some browsers.
+        it('should be possible to annotate ES6 methods',
+           function() { expect(annotate(eval('({ fn(x) { return; } })').fn)).toEqual(['x']); });
+
+        it('should be possible to instantiate ES6 classes', function() {
+          // Only Chrome (not even the FF we use) supports ES6 classes.
+          if (!/chrome/i.test(navigator.userAgent)) return;
+          providers('a', function() { return 'a-value'; });
+          var clazz = eval('(class { constructor(a) { this.a = a; } aVal() { return this.a; } })');
+          var instance = injector.instantiate(clazz);
+          expect(instance).toEqual({a: 'a-value'});
+          // Check if prototype is set up correctly.s
```

Comment:
Well, the call to `aVal()` would fail if the prototype was incorrect. But point taken, dropped the comment.